cmd/orchestration/app/controller_constructor.go

```diff
@@ -88,14 +87,7 @@ func (cc *ControllerConstructor) New(config *ctrl.Config, cctx *ctrl.Context) (*
 		Plugins:         cc.Plugins,
 		ClusterLocation: opts.Location.ClusterLocation(),
 		ClusterConfig:   toClusterConfig(opts.Cluster),
-		TagNames: wiring.TagNames{
```

Comment:
Why this change? Seems unrelated to LegacyConfig. Could you split into multiple PRs please?

Reply:
See the comment about introducing the tag generator. If the tags come from a function there's no need to make their keys come in via config.